packages/next/src/server/config.ts

```diff
@@ -1004,6 +1005,11 @@ function assignDefaults(
     ]),
   ]
 
+  if (!result.experimental.htmlLimitedBots) {
+    // @ts-expect-error: override the htmlLimitedBots with default string, type covert: RegExp -> string
```

Comment:
how come this can't be `new RegExp(HTML_LIMITED_BOT_URA_RE_STRING)`? Don't we do the `instanceof` check below to turn it into a string?

Reply:
It's the contract of `assignDefaults` which we assume the return value is NextComplete everywhere. In NextConfigComplete we've converted into a string for store purpose in manifest so it's a string.

Here the type of result.experimental is still `NextConfig.Experimental`